### ISSUE TITLE
[typescript-fetch] btoa used for encoding Basic auth doesn't work with multibyte characters

### ISSUE THREAD

**Issue by michael42** [swagger-api/swagger-codegen#7735]
##### Description

When trying to call an API with HTTP Basic security, if the username or password contains a character with a unicode code point >255, the browser (Chrome) throws a `DOMException: Failed to execute 'btoa' on 'Window': The string to be encoded contains characters outside of the Latin1 range.`.

That issue is documented by Mozilla [The Unicode Problem](https://developer.mozilla.org/en-US/docs/Web/API/WindowBase64/Base64_encoding_and_decoding#The_Unicode_Problem) and they also suggest a few solutions.

##### Swagger-codegen version

2.3.1, 2.3.0

##### Swagger declaration file content or url

```yaml
swagger: "2.0"
info:
  version: "1.0.0"
securityDefinitions:
  UserSecurity:
    type: basic
security:
  - UserSecurity: []
paths:
  /test:
    get:
      responses:
        "200": {}
```

##### Command line used for generation

`java -jar swagger-codegen-cli-2.3.1.jar generate -i test.yml -l typescript-fetch -o out/`

##### Steps to reproduce

* generate the code
* edit `api.ts` and insert a new line at the end:
  - `new DefaultApi({username: 't€stuser', password: 't€estpwd'}).testGet()`
* compile the code
* execute it in a typical browser
* an `Character Out Of Range` Error is thrown

##### Related issues/PRs

I didn't find anything, but a quick search for `btoa` suggests that it also affects the typescript-jquery and typescript-angular generators.

##### Suggest a fix/enhancement

As a quick workaround, I added the `b64EncodeUnicode` function from the MDN docs. A better fix would probably avoid using `btoa` altogether, as it isn't available out of the box inside Node.js and we currently work around that, too.

**Comment by BernhardPosselt** [comment]
Yeah including portable-fetch makes no sense if btoa is not available out of the box. 

Had to polyfill it on node using:

```js
if (!global.btoa) {
    global.btoa = require('btoa');
}
```
